### PR TITLE
Add 'minikube update' command

### LINKED ISSUE #2359: Do we have minikube update command to update to its latest version?
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
Kind of a question.

**What you expected to happen**:
I would like to update my minikube version to its latest by running `minikube update` command.

**Anything else do we need to know**:
Did we had any discussion before?

### PR DESCRIPTION
**Pull request by budhrg**
Fix #2359 

* also changes the behavior of update notification text to show update command instead of download URL

See demo https://asciinema.org/a/McxbtlFegfFIVf7zw73914MIG.

### CODE REVIEW

**Comment by budhrg** [comment]
Tested with Windows:
```
PS C:\> minikube.exe version
minikube version: v0.24.0

PS C:\> minikube.exe status
There is a newer version of minikube available (v0.25.0). Run 'minikube update'.

To disable this notification, run the following:
minikube config set WantUpdateNotification false
========================================
kubectl could not be found on your path. kubectl is a requirement for using minikube
To install kubectl, please do the following:

download kubectl from:
https://storage.googleapis.com/kubernetes-release/release/v1.9.0/bin/windows/amd64/kubectl.exe
Add kubectl to your system PATH

To disable this message, run the following:

minikube config set WantKubectlDownloadMsg false
========================================
minikube:
cluster:
kubectl:
PS C:\> minikube.exe update
Downloading latest minikube version v0.25.0
 41.53 MB / 41.53 MB [=====================================================================================================================================] 100.00% 0s
Updated successfully to version v0.25.0

PS C:\> minikube.exe version
minikube version: v0.25.0
```

**Comment by budhrg** [comment]
Anyone trying it? @r2d4 , @luxas or @aaron-prindle ?

**Comment by afbjorklund** [comment]
It would be great if you could log the actual error, and it also seems to be missing a newline ?

``` diff
diff --git a/cmd/minikube/cmd/update.go b/cmd/minikube/cmd/update.go
index 0b89dd3..a3993c4 100644
--- a/cmd/minikube/cmd/update.go
+++ b/cmd/minikube/cmd/update.go
@@ -36,19 +36,19 @@ var updateCmd = &cobra.Command{
 	Run: func(command *cobra.Command, args []string) {
 		currentVersion, err := version.GetSemverVersion()
 		if err != nil {
-			fmt.Fprint(os.Stderr, "Error fetching current version")
+			fmt.Fprint(os.Stderr, "Error fetching current version: %s\n", err)
 			os.Exit(1)
 		}
 
 		latestVersion, err := update.LatestVersion()
 		if err != nil {
-			fmt.Fprint(os.Stderr, "Error fetching latest version from internet")
+			fmt.Fprint(os.Stderr, "Error fetching latest version from internet: %s\n", err)
 			os.Exit(1)
 		}
 
 		if update.IsNewerVersion(currentVersion, latestVersion) {
 			if err := update.Update(latestVersion); err != nil {
-				fmt.Fprintf(os.Stderr, "Error updating to latest version: %s", err)
+				fmt.Fprintf(os.Stderr, "Error updating to latest version: %s\n", err)
 				os.Exit(1)
 			}
 
```

File: pkg/testing/http_mock.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	"time"
+)
+
+const (
+	OctetStream = "application/octet-stream"
+	JSON        = "application/json; charset=utf-8"
+	TEXT        = "text/plain"
+)
+
+var DefaultRoundTripper http.RoundTripper = &http.Transport{
```
Comment (dlorenc) [inline pkg/testing/http_mock.go]: Did you consider using https://golang.org/pkg/net/http/httptest/ instead of mocking at this level?

Comment (budhrg) [reply pkg/testing/http_mock.go]: Looking into it.

**Comment by dlorenc** [comment]
I'm goiong to close this as it's quite stale. Please reopen if you want to come back to it.

**Pull request closed without merging**